pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -171,18 +171,88 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, nil
 	}
 
+	// get the Namespace instance
+	ns := &corev1.Namespace{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, ns)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, it can be transitioning to the final desired state
+			// e. g. deletion or creation still in progress. Return and retry again
+			reqLogger.Info("Object not ready")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object
+		reqLogger.Info("Error Getting Namespace")
+		return reconcile.Result{}, err
+	}
+
+	// get the NamespaceList
+	nsList := &corev1.NamespaceList{}
+	opts = client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, nsList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get clusterRoleBindingList")
+		return reconcile.Result{}, err
+	}
+
+	// slice of clusterRoleName that does not exists as a clusterRole
+	permissionClusterRoleNameList := controllerutil.PopulateCrPermissionClusterRoleNames(instance, clusterRoleList)
+	for _, permissionClusterRoleName := range permissionClusterRoleNameList {
+		// update condition
+		updatedSubjectPermission := controllerutil.UpdateCondition(instance, permissionClusterRoleName+" for clusterPermission does not exist", permissionClusterRoleName, true, managedv1alpha1.SubjectPermissionFailed)
+		err = r.client.Status().Update(context.TODO(), updatedSubjectPermission)
+		if err != nil {
+			reqLogger.Error(err, "Failed to update condition.")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// compile list of allowed namespaces only for this subject permission. NOT a list of subject permissions
+	for _, permission := range instance.Spec.Permissions {
+		// list of all namespaces in safelist
+		safeList := controllerutil.GenerateSafeList(permission.NamespacesAllowedRegex, permission.NamespacesDeniedRegex, nsList)
+
+		// for each safelisted namespace
+		for _, ns := range safeList {
+			//create roleBinding
+			roleBinding := controllerutil.NewRoleBindingForClusterRole(permission.ClusterRoleName, instance.Spec.SubjectName, instance.Spec.SubjectKind, ns)
+			err := r.client.Create(context.TODO(), roleBinding)
```

Comment:
Do you need to check if that RB already exists in the namespace?

Reply:
added check for if RB exists in the namespace